docs/src/man/getting_started.md

```diff
@@ -444,6 +444,122 @@ julia> colwise(sum, df)
  10.0
 ```
 
+### Handling of Columns Stored in a `DataFrame`
+
+`DataFrame` assumes *ownership* of its columns, meaning
+that column vectors should generally not be mutated directly.
+
+In order to ensure the ownership principle functions that transform a `DataFrame` to produce a
+new `DataFrame` always perform a copy of the columns, for example:
+
+```jldoctest dataframe
+julia> df = DataFrame(A = 1:4, B = 4.0:-1.0:1.0)
+4×2 DataFrame
+│ Row │ A     │ B       │
+│     │ Int64 │ Float64 │
+├─────┼───────┼─────────┤
+│ 1   │ 1     │ 4.0     │
+│ 2   │ 2     │ 3.0     │
+│ 3   │ 3     │ 2.0     │
+│ 4   │ 4     │ 1.0     │
+
+julia> df2 = copy(df);
+
+julia> df2.A === df.A
+false
+```
+
+On the other hand, in-place functions, whose names end with `!`, may mutate the column vectors of the
+`DataFrame` they take as an argument, for example:
+
+```jldoctest dataframe
+julia> x = [3, 1, 2];
+
+julia> df = DataFrame(x=x)
+3×1 DataFrame
+│ Row │ x     │
+│     │ Int64 │
+├─────┼───────┤
+│ 1   │ 3     │
+│ 2   │ 1     │
+│ 3   │ 2     │
+
+julia> sort!(df)
+3×1 DataFrame
+│ Row │ x     │
+│     │ Int64 │
+├─────┼───────┤
+│ 1   │ 1     │
+│ 2   │ 2     │
+│ 3   │ 3     │
+
+julia> x
+3-element Array{Int64,1}:
+ 1
+ 2
+ 3
+
+julia> df.x[1] = 100
+100
+
+julia> df
+3×1 DataFrame
+│ Row │ x     │
+│     │ Int64 │
+├─────┼───────┤
+│ 1   │ 100   │
+│ 2   │ 2     │
+│ 3   │ 3     │
+
+julia> x
+3-element Array{Int64,1}:
+ 100
+   2
+   3
+```
+
+In-place functions are generally safe to call, except when a view of the `DataFrame` is in use
+(created via a `view`, `@view` or [`groupby`](@ref)). In the latter case, the view
+might become corrupted, which make trigger errors, silently return invalid data
+or even cause Julia to crash.
+
+It is possible to have a direct access to a column `col` of a `DataFrame` `df`
+using the syntaxes `df.col`, `df[:col]`, via the [`eachcol`](@ref) function or simply
+by storing the reference to the column vector before the `DataFrame` was created (note
+that in general the `DataFrame` constructor does not make copies).
+
+```jldoctest dataframe
+julia> x = [3, 1, 2];
+
+julia> df = DataFrame(x=x)
+3×1 DataFrame
+│ Row │ x     │
+│     │ Int64 │
+├─────┼───────┤
+│ 1   │ 3     │
+│ 2   │ 1     │
+│ 3   │ 2     │
+
+julia> df.x === x
+true
+
+julia> df[1] === x
+true
+
+julia> eachcol(df, false)[1] === x
+true
+```
+
+Note that a column obtained from a `DataFrame` using one of these methods should not be mutated because:
```

Comment:
This sounds a bit strong I think, since it's perfectly fine to mutate entries in a column in most cases (last bullet). What do you think about saying something like "should not be mutated without caution"?

Reply:
ok